File: fixtures/eval-agent-tasks/py-plugin-registry/app/registry.py
class PluginRegistry:
    def __init__(self):
        self._plugins = {}

    def register(self, name: str, factory):
        self._plugins[name] = factory
        return name

    def create(self, name: str):
        factory = self._plugins.get(name)
        return factory() if factory else None

File: fixtures/eval-l1-real/rust-linkme-plugins/src/registry.rs
//! Real-idiom multi-file: linkme distributed_slice registry (M3-E).

use linkme::distributed_slice;

pub struct PluginRegistration {
    pub name: &'static str,
    pub factory: fn() -> Box<dyn Fn() -> i32>,
}

pub struct MetricsPlugin;

impl MetricsPlugin {
    pub fn new() -> Box<dyn Fn() -> i32> {
        Box::new(|| 42)
    }
}

#[distributed_slice(PLUGINS)]
static METRICS: PluginRegistration = PluginRegistration {
    name: "metrics",
    factory: MetricsPlugin::new,
};
